libpod/container.go

```diff
@@ -906,12 +906,26 @@ func (c *Container) CgroupManager() string {
 	return cgroupManager
 }
 
-// CGroupPath returns a cgroups "path" for a given container.
+// CGroupPath returns a cgroups "path" for the given container.
+// Note that the container must be running.  Otherwise, an error
+// is returned.
 func (c *Container) CGroupPath() (string, error) {
 	if c.config.NoCgroups || c.config.CgroupsMode == "disabled" {
 		return "", errors.Wrapf(define.ErrNoCgroups, "this container is not creating cgroups")
 	}
 
+	if !c.batched {
+		c.lock.Lock()
+		defer c.lock.Unlock()
+		if err := c.syncContainer(); err != nil {
+			return "", errors.Wrapf(err, "error updating container %s state", c.ID())
```

Comment:
Would it be better to add to or replace the message with the action we're taking here?  Perhaps:
```suggestion
			return "", errors.Wrapf(err, "error updating container %s state when retrieving the path for CGroup ", c.ID())
```

Reply:
@mheon WDYT? AFAICS this error message is used across many places. But I do see wisdom in customizing them a bit to discriminate one from another.

Reply:
I don't mind - it doesn't hurt - but if Go was a more expressive language, this entire block of code would be a macro so we wouldn't have to copy and paste everywhere.

Reply:
(In which case, it really would be guaranteed to be the same error message everywhere...)